### DISCUSSION TITLE
Vitest and Pinia reactivity not working when running multiple tests

### DISCUSSION THREAD

**Discussion by wasimTQ** [vuejs/pinia#2255] [Help and Questions]
I'm testing my custom store built with `createGlobalState` from `vueuse`. I'm using a watcher to watch for changes in pinia's getter to manipulate value in my store.

For some context: 
- `cardExpandedDialogStore` is a pinia store
- `cardNavigation` is a custom store
- `cardExpandedDialogStore.open()` when I call this function it modifies the store values `dialogIsVisible, expandedCard` that's used inside the custom store to manipulate values.

The problem is it works perfect when I'm only running that test suite with `it.only`. And not working when running multiple test suites.

When running the single test suite alone, the response for 
``console.log(currentIndex.value, currentCardId.value) // These are from cardNavigation`` is `1 'd66ee8a8-c650-42b7-bbe0-7c8761473b45'`.

But when running multiple test suites other tests are passing but only two tests that calls the `open` function is failing and the console log of the above in `0 `

I know I should've explained it better. But I don't know how. And I can't share much of codes that's protected by NDA. Let me know if you can help me solve it.

**Comment by posva** [comment]
This seems to be related to vueuse, not pinia. BTW you shouldn't use a global state with pinia as pinia already makes its state global. I suspect the global state you are creating is being reused across tests. With pinia, each each test should have its own `createPinia()` to ensure they run with their own fresh state https://pinia.vuejs.org/cookbook/testing.html#testing-stores

**Reply by wasimTQ** [reply]
Yeah I have it set up like this in `beforeEach`. I couldn't really point out a problem there.

I also reset the stores I'm using in `afterEach`.

The pinia's reactivity is updated but it doesn't seems to work on the custom composable.

**Reply by wasimTQ** [reply]
Seems to be not working in some instance. So I worked up another work around.

I thought of do it like integration testing. But mocked the function from the custom store instead.

**Reply by posva** [reply]
Instead of resetting stores, use a new pinia instance like shown in docs
